### DISCUSSION TITLE
Two questions about new ASReview

### DISCUSSION THREAD

**Discussion by Rinus-Verdonschot-UB** [asreview/asreview#2349] [Q&A]
Dear all, 

I have two questions. Thanks in advance for your answer.

#1: If a person screening with ASReview has completed SAFE 2, the only thing s/he needs to do to train the more complicated model is to go from Elas-U4 to Elas-H3 (after installing asreview-dory of course) and then wait until the model is trained before continuing?

#2: If one wants to do Step 4 of the SAFE procedure, one possible way would be to use the 10 highest and 10 lowest ranked articles on the final Step 3 set on a new model (if I understood it correctly). However, if one clicks "export" (to excel). Do we get the ranking of the papers which have a decision. It seems now that

a) we get all 1 labeling decisions in the order they were presented
b) we get the unlabeled articles ranked.
c) we get all 0 labeling decisions in the order they were presented

What I would like is the order of the last iteration based on the current ranking (so not on presentation order for 0, 1 labels).. If you do simulations you can get this with the column "labeling_order" (which is present in the .asreview file). Is such a thing possible with ASReview 2.1 lab?

Oh and what does "id" mean in the output file? (bonus question)

<img width="1607" height="104" alt="image" src="https://github.com/user-attachments/assets/82fb3ece-39ec-4177-a737-1127c96fac75" />

Once again, thanks for answering my questions.

Rinus Verdonschot
University Library Maastricht

**Comment by Rinus-Verdonschot-UB** [comment]
Anybody??

**Comment by lottewerf** [comment]
Hey everyone,

I would also be very interested to know the answers to these questions!
Is there anyone who could help us?

Lotte van der Werf
Maastro Clinic, the Netherlands

**Comment by Rensvandeschoot** [comment]
Question no/1: yes, indeed! see our [recent paper](https://doi.org/10.1016/j.patter.2025.101318) for more information, or go to the [documentation](https://asreview.readthedocs.io/en/stable/lab/models.html)

Question no/2: The order of the dataset is available in the [documentation](https://asreview.readthedocs.io/en/stable/lab/progress.html#variables-in-the-exported-dataset). For the Safe procedure, in my opinion, it does not matter too much what you use as training data, as long as you are transparant about your choices. In v1 you needed both relevant and irrelevant lables to train a model, but in v2 you can start with only a set of relevant labels. So, your could also export only the relevant ones. I would recommend to use the Full Text decissions in this step. 

Bonus Question: All variables are explained in our documentation (okay, I have to confess that I also need to search the hundreds of pages), but [here](https://asreview.readthedocs.io/en/stable/technical/reference/asreview.Record.html#asreview-record) you can find the relevant info, but in short: record_id is an internal identifier to connect rows in the dataset to the labels. I am actually not sure why it is exported... (sorry)

**Comment by Rinus-Verdonschot-UB** [comment]
Hi Rens,

Thank you for answering the question. For my question #2, this has more to do with Phase 4 of the SAFE procedure.

<img width="413" height="299" alt="image" src="https://github.com/user-attachments/assets/f19afa1c-e499-4b57-bcec-28022f8cd94c" />

"To train the active learning model, **the 10 highest- and lowest-ranked records** from the previous phase can be used"

Now we get as output (taken from the documentation).

All relevant records you have seen in the order they were shown during the screening process.
All records not seen during the screening and ordered from most to least relevant according to the last iteration of the model.
All non-relevant records are presented in the order these are shown during the screening process.

But wouldnt we need the final ranking when we stopped **of all seen records in the ranked order**? Instead of the seen records (relevant/irrelevant) in the order they were presented? or does the order they were presented in **become** the ranked order? or is it that an article I have already deemed "relevant" still shifts position in the rank depending on the other judgments? (which I believed it did).

Sorry if its an obvious answer and/or if I am slow in understanding it, I just want to make very clear how we can get these 10 highest and lowest ranked records out according to phase 4. Thanks once again for your advice.

Kind regards,

Rinus

**Reply by Rensvandeschoot** [reply]
In the paper by [Lombaers et al](https://www.mdpi.com/2076-3417/14/9/3842), we explain which information about the rank ordering is stored in what phase of the iterations. But, being more pragmatic, the last step is to double-check the exclusions, and for the model, you need training data. In the SAFE paper, we provided one suggestion on selecting training data, but there are 1001 other options. The fourth step is not about the training data but screening the excluded papers. So, I suggest not taking the statement about selecting the 10 papers too strictly, but interpreting it as a suggestion and deciding on your own procedure. I would do with version 2 of ASReview, only use all the relevant papers from the earlier phase, and do not pre-select any irrelevant papers (this option was simply impossible in v1). Hope this helps!
